### PR TITLE
feat: content security policy

### PR DESCRIPTION
**Pull request by barnettx**
Add content security policy (CSP) using helmet middleware to reduce chances of XSS attacks: https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP. Implementation based off GoGovSG's CSP.

**New dependencies**:
- Helmet

### CODE REVIEW

File: src/server/bootstrap/helmet.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+const cspOnlyReportViolations = config.get('cspOnlyReportViolations')
+const cspReportUri = config.get('cspReportUri')
+
+export default helmet({
+  contentSecurityPolicy: {
+    directives: {
+      defaultSrc: ["'self'"],
+      styleSrc: ["'self'", "'unsafe-inline'", 'https://fonts.googleapis.com/'],
+      fontSrc: ["'self'", 'https://fonts.gstatic.com/'],
+      imgSrc: ["'self'", 'data:'],
```
Comment (barnettx) [inline src/server/bootstrap/helmet.ts]: `'data:'` from Go. Don't think we have inline base64 images

File: src/server/bootstrap/helmet.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import config from '../config'
+
+const cspOnlyReportViolations = config.get('cspOnlyReportViolations')
+const cspReportUri = config.get('cspReportUri')
+
+export default helmet({
+  contentSecurityPolicy: {
+    directives: {
+      defaultSrc: ["'self'"],
+      styleSrc: ["'self'", "'unsafe-inline'", 'https://fonts.googleapis.com/'],
```
Comment (barnettx) [inline src/server/bootstrap/helmet.ts:13]: need to investigate further whether we use inline styles

File: src/server/bootstrap/helmet.ts (hunk 1 above)
Comment (LoneRifle) [reply src/server/bootstrap/helmet.ts]: Indeed we don't, so you can remove this.

File: src/server/bootstrap/helmet.ts (hunk 2 above)
Comment (LoneRifle) [reply src/server/bootstrap/helmet.ts:13]: Mostly by React

File: src/server/bootstrap/helmet.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+const cspReportUri = config.get('cspReportUri')
+const sentryDns = config.get('sentryDns')
+
+export default helmet({
+  contentSecurityPolicy: {
+    directives: {
+      defaultSrc: ["'self'"],
+      styleSrc: ["'self'", "'unsafe-inline'", 'https://fonts.googleapis.com/'],
+      fontSrc: ["'self'", 'https://fonts.gstatic.com/'],
+      connectSrc: ["'self'", ...(sentryDns ? [parseDomain(sentryDns)] : [])],
```
Comment (barnettx) [inline src/server/bootstrap/helmet.ts:15]: maybe disable the linting here so that sentryDns is on a new line for better readibility?

Comment (LoneRifle) [reply src/server/bootstrap/helmet.ts:15]: This is fine

File: src/server/bootstrap/helmet.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      ...(cspReportUri ? { reportUri: cspReportUri } : {}),
```
Comment (LoneRifle) [inline src/server/bootstrap/helmet.ts:16]: You didn't put the report Uri in an array in the end?

**Review by LoneRifle — CHANGES_REQUESTED** [review]
Lgtm otherwise, but can you also enable CSP for the handler at `src/serverless/static.ts`? 

Once merged, push to `staging`, then verify that the changes work as intended.

File: src/server/bootstrap/helmet.ts (hunk 4 above)
Comment (barnettx) [reply src/server/bootstrap/helmet.ts:16]: In Sentry's example for how to set up report-uri, they don't include report-uri in the connect-src directive
```
def middleware(request, response):
    response['Content-Security-Policy-Report-Only'] = \
        "default-src 'self'; " \
        "report-uri https://o545501.ingest.sentry.io/api/5667596/security/?sentry_key=5049a239ada645f5900b090068b008bf"
    return response
```

I also tested it with a personal sentry i set up. The CSP reports go through.
image

Comment (barnettx) [reply src/server/bootstrap/helmet.ts:16]: In Sentry's example for how to set up report-uri, they don't include report-uri in the connect-src directive
```
def middleware(request, response):
    response['Content-Security-Policy-Report-Only'] = \
        "default-src 'self'; " \
        "report-uri https://o545501.ingest.sentry.io/api/5667596/security/?sentry_key=XXXXXXXXXXXXXXXXX"
    return response
```

I also tested it with a personal sentry i set up. The CSP reports go through.
image

**Review by LoneRifle — APPROVED** [review]

**Comment by barnettx** [comment]
reminder that CSP only works on local development when using server to serve front-end. webpack dev server does not use the helmet CSP
```
npm run build
npm run dev-server
```
then view access the app at the server port (8080)

**Pull request merged**